### PR TITLE
JBPM-8531: Process Instance Variable 'initiator' editing unexpected e…

### PR DESCRIPTION
**Pull request by nmirasch**
…rror
add deploymentId and serverTemplateId to all ProcessInstanceVariables, even they are not at the process definition

### CODE REVIEW

**Comment by nmirasch** [comment]
@cristianonicolai @jstastny-cz could you take a look? Thanks!

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java [hunk 1]
```diff
@@ -80,7 +80,16 @@ public static void registerProcessor(VariableProcessor processor) {
 
         if (!filteredVariables.isEmpty()) {
             filteredVariables.forEach(variable -> {
-                variablesSummary.add(new ProcessVariableSummary(variable.getVariableName(), variable.getVariableName(), variable.getProcessInstanceId(), variable.getOldValue(), variable.getValue(), variable.getDate().getTime(), ""));
+                ProcessVariableSummary variableSummary = new ProcessVariableSummary(variable.getVariableName(),
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java]: I would suggest using the constructor to pass all variables and also transform this into a stream style, using filteredVariables.map. The map function can be an isolated function to be tested independently.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                                          .name(processDefaultExcludedVar_name).value(processId)
+                                          .date(new Date()).build());
+        properties = new HashMap<>();
+        properties.put(processDefinitionVar1_Name, "String");
+        properties.put(processDefinitionVar2_Name, "String");
+        properties.put(processDefinitionVar3_Name, "String");
+    }
+
+    @Test
+    public void AddDeploymentId_ServerTemplateToAllProcessInstancesVarsTest() {
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: change name to camelCase ... i.e. lower case first letter and remove underscore.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    private String processDefinitionVar1_Name = "var1";
+    private String processDefinitionVar2_Name = "var2";
+    private String processDefinitionVar3_Name = "var3";
+    private String processDefaultVar_name = "initiator";
+
+    private String processDefinitionVar1_Value = "value1";
+    private String processDefaultVar_Value = "testUser";
+    private String processDefaultExcludedVar_name = "processId";
+    List<VariableInstance> variables;
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: Is there any reason for these two fields not to be private?

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+@RunWith(MockitoJUnitRunner.class)
+public class VariableHelperTest {
+
+    private final String deploymentId = "deploymentId";
+    private final String serverTemplateId = "serverTemplateId";
+    private final String processId = "processId";
+    private final Long processInstanceId = 1L;
+
+    private String processDefinitionVar1_Name = "var1";
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: remove the underscores from the field names.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java (hunk 1 above)
Comment (nmirasch) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/util/VariableHelper.java]: oks

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java (hunk 2 above)
Comment (nmirasch) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: ok

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java (hunk 4 above)
Comment (nmirasch) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: oks

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java (hunk 3 above)
Comment (nmirasch) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: nop!

**Comment by nmirasch** [comment]
@cristianonicolai, @jstastny-cz Thanks! code updated, I just have created the ProcessInstanceVariableMapper to encapsulate that mapping and fixed some naming problems following your suggestion, Could you give to this another try?

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                                          .name(processDefaultExcludedVarName).value(processId)
+                                          .date(new Date()).build());
+        properties = new HashMap<>();
+        properties.put(processDefinitionVar1Name, "String");
+        properties.put(processDefinitionVar2Name, "String");
+        properties.put(processDefinitionVar3Name, "String");
+    }
+
+    @Test
+    public void AddDeploymentIdServerTemplateToAllProcessInstancesVarsTest() {
```
Comment (jstastny-cz) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: method name should start with lower case letter

Comment (nmirasch) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/util/VariableHelperTest.java]: ups!
@jstastny-cz Updated, Thanks

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

**Comment by nmirasch** [comment]
rebased!

**Comment by nmirasch** [comment]
Jenkins execute full downstream build

**Comment by nmirasch** [comment]
rebased

**Comment by nmirasch** [comment]
ok to test

**Pull request merged**